### PR TITLE
assigment-1: Implemented basic and additional assigment, added tests …

### PR DESCRIPTION
**Pull request by monikeu**
…and readme

### CODE REVIEW

File: monikeu/assignment-1/src/main/java/AnagramEngine.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    private Map<Character, Integer> countLetters(String string) {
+
+        Map<Character, Integer> lettersCount = new HashMap<>();
+
+        for (int i = 0; i < string.length(); i++) {
+            lettersCount.put(string.charAt(i), 0);
+        }
+
+        for (int i = 0; i < string.length(); i++) {
```
Comment (TDiva) [inline monikeu/assignment-1/src/main/java/AnagramEngine.java]: Do we really need to have two loops here?
I would be nicer to have a single one which does all computations! Or maybe even treat string as a stream of characters?

File: monikeu/assignment-1/src/main/java/SentencesSolver.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        return true;
+    }
+
+    private String[] getWords(String string1) {
+        return removeUnneededCharacters(string1).split("\\s+");
+    }
+
+
+    private String removeUnneededCharacters(String string1) {
+        for (int i = 0; i < unnededCharacters.length; i++) {
```
Comment (TDiva) [inline monikeu/assignment-1/src/main/java/SentencesSolver.java]: Better using "for each" loop here, not to get lost in indicies.

File: monikeu/assignment-1/src/test/java/WithSpecialCharactersSolverTest.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+public class WithSpecialCharactersSolverTest {
+    private static WithSpecialCharactersSolver withSpecialCharactersSolver;
+
+    @BeforeClass
+    public static void setUp() {
+        withSpecialCharactersSolver = new WithSpecialCharactersSolver();
+    }
+
+
```
Comment (TDiva) [inline monikeu/assignment-1/src/test/java/WithSpecialCharactersSolverTest.java]: Please, remove these empty lines, there are too many.

Comment (monikeu) [reply monikeu/assignment-1/src/test/java/WithSpecialCharactersSolverTest.java]: This comment made me rethink all of my empty lines ans styling, thanks.

File: monikeu/assignment-1/src/main/java/AnagramEngine.java [hunk 4]
```diff
@@ -1,36 +1,24 @@
...
-            lettersCount.put(string.charAt(i), 0);
-        }
-
-        for (int i = 0; i < string.length(); i++) {
-            lettersCount.put(string.charAt(i), lettersCount.get(string.charAt(i)) + 1);
-        }
-
-        return lettersCount;
+    private HashMap<Integer, Long> countLetters(String string) {
+        return string.chars().boxed().collect(Collectors.groupingBy(id -> id, HashMap::new, Collectors.counting()));
```
Comment (TDiva) [inline monikeu/assignment-1/src/main/java/AnagramEngine.java]: Wonderful!

**Pull request merged**